fix(validation): improve error messages for invalid username

Riddle-Me-Sid opened the pull request:
## Changes
In reference to Feature Request:
https://github.com/dailydotdev/daily/issues/1773

- Updated the error message for invalid characters in usernames 
    -  New Invalid Character Error
    ```Invalid character detected! Only underscores (_) allowed```
    -  New Invalid length Error
    ```Username must be between 3-38 characters```
- Added handling for minimum and maximum length.
<!--
### Describe what this PR does

- Short and concise, bullet points can help
- Screenshots if applicable can also help
-->

## Events

Did you introduce any new tracking events?
No

<!--
If yes please remove the comment HTML comment tags and fill the table below

Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

-->

## Experiment

Did you introduce any new experiments?
No
<!--
If yes please remove the comment HTML comment tags and follow the instructions below

Don't forget to send a message to the [#experiments](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715175315620999) channel, following the template in slack, and adding a link to the message here.

> [!IMPORTANT]
> Please do not merge the PR until the experiment enrolment is approved.

-->

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

I was unable to get the GItpod instance working because of errors launching the API. So I wasn't able to do testing in the actual environment.

<!--
If relevant, please remove the comment HTML comment tags and fill the checkboxes below

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

-->

<!--
If the branch name does not beging with a jira ticket, please copy and paste the
below line outside the HTML comment tags to link this PR to the ticket in Jira.

AS-{number}
or
MI-{number}
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dailydotdev/apps?pullRequest=4284) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dailydotdev/apps?pullRequest=4284) before we can accept your contribution.<br/><hr/>**riddle-me-sid** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dailydotdev/apps?pullRequest=4284) it.</sub>

sshanzel reviewed (approved):
Great catch! Definitely helps user figuring out what's allowed and what's not 🙏

sshanzel commented on packages/shared/src/hooks/useProfileForm.ts:
```diff
@@ -88,6 +88,10 @@ export const onValidateHandles = (
+    if (after.username.length > 38) {
+      return { username: errorMessage.profile.usernameLength };
+    }
+    
```
```suggestion

```

sshanzel commented:
One thing I missed before merging the PR, can you complete this step? Just some docs about being a contributor in dailydev 🙏

https://cla-assistant.io/dailydotdev/apps?pullRequest=4284

Riddle-Me-Sid commented:
@sshanzel I am sorry, I did sign it but it's not seeing it. I realize I accidentally had my Git email configured to riddle.me.sid@example.com instead of gmail.com. I have fixed that, but I am not sure if there is a way to update it on this PR without resubmitting it.

I added the wrong address to my github, but it doesn't seem like the CLA assistant is picking that up. Do you know a way around that, or would you like me to submit another pull request to make sure the CLA is linked properly?
